timed/tracking/models.py

```diff
@@ -105,11 +106,16 @@ def save(self, *args, **kwargs):
 
         This rounds the duration of the report to the nearest 15 minutes.
         However, the duration must at least be 15 minutes long.
+
+        Sets the billed state to the billed state of the project.
         """
         self.duration = timedelta(
             seconds=max(15 * 60, round(self.duration.seconds / (15 * 60)) * (15 * 60))
         )
 
+        if not self.pk or not self.billed:
+            self.billed = self.task.project.billed
```

Comment:
This marks all newly created reports as `billed` if the project is `billed`.
What about existing reports?
* Should they get marked as `billed` if the project is updated (set to `billed=True`)?
In that case we would need to overwrite the `Project.save()` method too.

* Should they get marked as `billed` if they are updated after `project` is updated?

Reply:
1. It wasn't specified to do that automatically, so no.
2. Yes, if they get moved or updated on a billed project it should also mark them

Reply:
Alright, for 2 I'm pretty sure this doesn't work yet. Please add a test case where the project is set as billed and an update on a report (text field or so) yields a `billed=True`.

```suggestion
        if not self.billed:
            self.billed = self.task.project.billed
```

Reply:
Added the test and changed it so it updates the billed flag